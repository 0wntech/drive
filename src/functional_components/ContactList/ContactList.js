import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './ContactList.module.scss';
import { ContactListItem } from '../ContactListItem';

const ContactList = ({
    contacts,
    onItemClick,
    webId,
    removeContact,
    addContact,
    isContact,
    removable,
}) => {
    const renderContact = (contact) => {
        return (
            <ContactListItem
                addContact={() => addContact(webId, contact)}
                onClick={(contact) => {
                    onItemClick(contact);
                }}
                contact={contact}
                removeContact={() => removeContact(webId, contact)}
                isContact={isContact}
                removable={removable}
            />
        );
    };

    return (
        <div
            className={classNames(styles.container, {
                [styles.empty]: !contacts || contacts.length === 0,
            })}
        >
            {contacts && contacts.length > 0 ? (
                contacts.map((contact) => (
                    <div key={contact.webId} className={styles.itemContainer}>
                        {renderContact(contact)}
                    </div>
                ))
            ) : (
                <div className={styles.emptyMessage}>
                    You don't have any contacts yet. Try searching for a contact
                    through the search bar.
                </div>
            )}
        </div>
    );
};

ContactList.propTypes = {
    contacts: PropTypes.arrayOf(PropTypes.object),
    onItemClick: PropTypes.func,
    webId: PropTypes.string,
    removeContact: PropTypes.func,
};

export default ContactList;
