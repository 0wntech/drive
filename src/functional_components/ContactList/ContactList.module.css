.container {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    width: 100%;
}

.itemContainer {
    width: calc(50% - 10px);
    margin-bottom: 20px;
}
