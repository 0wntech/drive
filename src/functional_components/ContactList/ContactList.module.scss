@import '../../styles/constants.scss';

.container {
    display: flex;
    flex-flow: row wrap;
    align-items: flex-start;
    width: 100%;
    min-height: 380px;
}

.empty {
    justify-content: center;
}

.emptyMessage {
    align-self: center;
    text-align: center;
    font-size: 2em;
    width: 50%;
}

.itemContainer {
    width: calc((100% - 40px) / 5);
}

.itemContainer + .itemContainer {
    margin-left: 10px;
    margin-bottom: 10px;
}

.itemContainer:nth-child(6) {
    margin-left: initial;
}

@media (max-width: $screen_l) {
    .itemContainer {
        width: 100%;
        margin: initial;
    }

    .itemContainer + .itemContainer {
        margin: initial;
        margin-top: 10px;
    }
}
