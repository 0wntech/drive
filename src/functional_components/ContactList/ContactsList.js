import React from 'react';
import PropTypes from 'prop-types';
import styles from './ContactList.module.css';
import { ContactListItem } from '../ContactListItem';

const ContactList = ({ contacts }) => {
    return (
        <div className={styles.container}>
            {contacts
                ? contacts.map((contact) => (
                      <div className={styles.itemContainer}>
                          <ContactListItem contact={contact} />
                      </div>
                  ))
                : null}
        </div>
    );
};

ContactList.propTypes = {
    contacts: PropTypes.arrayOf(PropTypes.object),
};

export default ContactList;
