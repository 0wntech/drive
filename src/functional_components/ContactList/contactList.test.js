import React from 'react';
import TestRenderer from 'react-test-renderer';
import ContactList from './ContactList';
import { contacts } from '../../../test/fixtures/contacts';

describe('ContactList', () => {
    test('should render correctly', () => {
        const tree = TestRenderer.create(<ContactList contacts={contacts} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
