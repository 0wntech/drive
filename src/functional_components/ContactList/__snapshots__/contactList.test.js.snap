// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="itemContainer"
  >
    <div
      className="container"
      onClick={[Function]}
    >
      <div
        className="imageContainer"
        data-test-id="contact-picture"
        onClick={[Function]}
      >
        <div
          className="image"
          style={
            Object {
              "backgroundImage": "url('https://ludwigschubert.solid.community/profile/aa.jpeg')",
            }
          }
        />
      </div>
      <div
        className="nameContainer"
      >
        Ludwig Faust Schubert
      </div>
      <div
        className="iconContainer added"
      />
    </div>
  </div>
  <div
    className="itemContainer"
  >
    <div
      className="container"
      onClick={[Function]}
    >
      <div
        className="imageContainer"
        data-test-id="contact-picture"
        onClick={[Function]}
      >
        <div
          className="image"
          style={
            Object {
              "backgroundImage": "url('https://bejow.solid.community/profile/meme_ben_gang.jpg')",
            }
          }
        />
      </div>
      <div
        className="nameContainer"
      >
        Ben Wetzel
      </div>
      <div
        className="iconContainer added"
      />
    </div>
  </div>
  <div
    className="itemContainer"
  >
    <div
      className="container"
      onClick={[Function]}
    >
      <div
        className="imageContainer"
        data-test-id="contact-picture"
        onClick={[Function]}
      >
        <div
          className="image"
          style={
            Object {
              "backgroundImage": "url('https://bejow.solid.community/profile/meme_ben_gang.jpg')",
            }
          }
        />
      </div>
      <div
        className="nameContainer"
      >
        Ben Wetzel
      </div>
      <div
        className="iconContainer added"
      />
    </div>
  </div>
</div>
`;
